oc-chef-pedant/spec/api/compliance_proxy_spec.rb

```diff
@@ -1,43 +1,73 @@
+def wait_for_http_listener(uri_string, retries = 10)
+  require 'net/http'
+  require 'openssl'
+  require 'uri'
+  uri = URI(uri_string)
+  begin
+    puts "Trying GET #{uri_string}"
+    Net::HTTP.start(uri.host, uri.port, :use_ssl => true,
+                    verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
+      request = Net::HTTP::Get.new uri
+      http.request request
+    end
+    puts "Succesfully made GET to #{uri_string}"
+  rescue StandardError => e
+    if retries <= 0
+      raise e
+    else
+      puts "GET #{uri_string} failed: #{e} retrying in 1 seconds (#{retries} retries remaining)"
+      retries -= 1
+      sleep 1
+      retry
+    end
+  end
+end
+
+def start_compliance_stub(port)
+  # Start stub compliance server in a thread
+  require 'webrick'
+  require 'webrick/https'
+  require 'openssl'
+  require 'thread'
+  Thread.new do
+    server = WEBrick::HTTPServer.new(:Port => port,
+                                     :SSLEnable => true,
+                                     :SSLVerifyClient => OpenSSL::SSL::VERIFY_NONE,
+                                     :SSLCertName => [%w[CN localhost]])
+    server.mount_proc '/' do |req, res|
+      if req.path =~ %r{/compliance/profiles/([^/]+)/?}
```

Comment:
[nit] the `if` and the `elsif` branch look like they could possibly be combined. But this is fine 👍